.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,26 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  schedule:
+    - cron: '0 0 * * 1'
```

Comment:
Why not run it on every pull request?  I'm not familiar with this tool so I don't know what the best practices are.